### PR TITLE
Fix : Pressing Tab button in AddressSearch inputs

### LINKED ISSUE #6235: [HOLD for payment 2022-02-01] Pressing "Tab" key while the result list of AddressSearch is open leaves it open even after clicking outside
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Type an address that yield results
2. Press "Tab" key
3. Click somewhere else in the form

## Expected Result:
The list of results should close after clicking outside
Focused element of the list after pressing tab should be highlighted

## Actual Result:
The list remains open event after focusing other inputs
No element of the list gets highlighted even if the focus changed to one of them after pressing tab

https://user-images.githubusercontent.com/87341702/140454912-b4a5d533-105c-48ff-9dcf-68e491dc7058.mov

## Workaround:
The user can still close the list by focusing again on the AddressSearch input and clicking outside. Or the user can just ignore the opened list.

## Platform:

Where is this issue occurring?

- Web
- Desktop App

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

### PR DESCRIPTION
**Pull request by developvsn**
### Fixed Issues

$ [6235](https://github.com/Expensify/App/issues/6235)

### Tests | QA Steps

1.Navigate to Company Information window.
2.Type an address that yield results in address search input.
3.Press "Tab" key or click to other input.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/71173662/142173855-79087833-9e9b-4701-a6e3-8495cd583587.mov

#### Desktop

https://user-images.githubusercontent.com/71173662/142173889-2e2cbbb8-f19b-4286-bbeb-706060472460.mov

### CODE REVIEW

**Comment by developvsn** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/components/AddressSearch.js [hunk 1]
```diff
@@ -86,10 +87,15 @@ const AddressSearch = (props) => {
         <GooglePlacesAutocomplete
             ref={googlePlacesRef}
             fetchDetails
+            onBlur={() => {
+                if(!isSelected){
```
Comment (NikkiWines) [inline src/components/AddressSearch.js:91]: This doesn't conform to our style guide - see [here](https://github.com/Expensify/App/blob/main/STYLE.md#whitespace) for more details around whitespace usage.

File: src/components/AddressSearch.js [hunk 2]
```diff
@@ -109,9 +115,15 @@ const AddressSearch = (props) => {
                 label: props.label,
                 containerStyles: props.containerStyles,
                 errorText: props.errorText,
+                onKeyPress: (event) => {
+                    if(event.key == 'Tab' && !isSelected){
```
Comment (NikkiWines) [inline src/components/AddressSearch.js:119]: Whitespace modifications needed here as well

Also this should be `===`

File: src/components/AddressSearch.js [hunk 3]
```diff
@@ -109,9 +115,15 @@ const AddressSearch = (props) => {
...
+                        googlePlacesRef.current.setAddressText("")
```
Comment (NikkiWines) [inline src/components/AddressSearch.js:120]: Missing semicolon here

File: src/components/AddressSearch.js [hunk 4]
```diff
@@ -109,9 +115,15 @@ const AddressSearch = (props) => {
...
+                    }
+                },
                 onChangeText: (text) => {
                     const isTextValid = !_.isEmpty(text) && _.isEqual(text, props.value);
-
+                    setIsSelected(false)
```
Comment (NikkiWines) [inline src/components/AddressSearch.js:125]: Missing semicolon here

File: src/components/AddressSearch.js [hunk 5]
```diff
@@ -86,10 +87,15 @@ const AddressSearch = (props) => {
         <GooglePlacesAutocomplete
             ref={googlePlacesRef}
             fetchDetails
+            onBlur={() => {
+                if(!isSelected){
+                    googlePlacesRef.current.setAddressText("")
```
Comment (NikkiWines) [inline src/components/AddressSearch.js:92]: Missing semicolon here

File: src/components/AddressSearch.js [hunk 6]
```diff
@@ -86,10 +87,15 @@ const AddressSearch = (props) => {
...
+                }}}
```
Comment (NikkiWines) [inline src/components/AddressSearch.js:93]: one of these `}` should be on a new line

File: src/components/AddressSearch.js [hunk 7]
```diff
@@ -86,10 +87,15 @@ const AddressSearch = (props) => {
...
             suppressDefaultStyles
             enablePoweredByContainer={false}
             onPress={(data, details) => {
                 saveLocationDetails(details);
+                setIsSelected(true)
```
Comment (NikkiWines) [inline src/components/AddressSearch.js:98]: missing semicolon here

**Review by NikkiWines — CHANGES_REQUESTED** [review]
Needs some minor adjustments, mainly stylistic.

File: src/components/AddressSearch.js [hunk 8]
```diff
@@ -86,10 +87,18 @@ const AddressSearch = (props) => {
         <GooglePlacesAutocomplete
             ref={googlePlacesRef}
             fetchDetails
+            onBlur={() => {
+                if (isSelected) {
+                    return;
+                }
+
```
Comment (NikkiWines) [inline src/components/AddressSearch.js:94]: Extra line

File: src/components/AddressSearch.js [hunk 9]
```diff
@@ -86,10 +87,17 @@ const AddressSearch = (props) => {
         <GooglePlacesAutocomplete
             ref={googlePlacesRef}
             fetchDetails
+            onBlur={() => {
+                if (isSelected) {
```
Comment (NikkiWines) [inline src/components/AddressSearch.js:91]: From the travis failure it looks like we prefer an early return here -- https://github.com/Shopify/web-configs/blob/main/packages/eslint-plugin/docs/rules/prefer-early-return.md

**Review by NikkiWines — CHANGES_REQUESTED** [review]

File: src/components/AddressSearch.js [hunk 10]
```diff
@@ -82,14 +83,30 @@ const AddressSearch = (props) => {
         }
     };
 
+    const checkPressingTab = (event) => {
+        if (event.key !== 'Tab') {
```
Comment (NikkiWines) [inline src/components/AddressSearch.js:87]: hmm, I don't think this needs to be it's own variable. Why not do the following? 
```Javascript
onKeyPress: (event) => {
    if (event.key !== 'Tab' || isSelected) {
        return;
    }
    googlePlacesRef.current.setAddressText('');
},
```

**Review by NikkiWines — CHANGES_REQUESTED** [review]

File: src/components/AddressSearch.js (hunk 10 above)
Comment (developvsn) [reply src/components/AddressSearch.js:87]: We tried similar way, but with "&&" operator instead "||", which led to the fact that only the last entered character would fit in the text field. Thank you.

**Review by NikkiWines — APPROVED** [review]
👍

**Comment by parasharrajat** [comment]
@NikkiWines These commits are not signed.

**Comment by NikkiWines** [comment]
Nice catch @parasharrajat, I completely missed that (I've become too reliant on our automated checks in other repos 😅) 

Let's wait on consensus from Cole in [this Slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1637258565311200?thread_ts=1637250519.303700&cid=C01GTK53T8Q), but I think Brandon's suggestion to revert + re-merge with signed commits is a good idea.

**Comment by developvsn** [comment]
> Nice catch @parasharrajat, I completely missed that (I've become too reliant on our automated checks in other repos 😅)
> 
> Let's wait on consensus from Cole in [this Slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1637258565311200?thread_ts=1637250519.303700&cid=C01GTK53T8Q), but I think Brandon's suggestion to revert + re-merge with signed commits is a good idea.

Nikki can you help with slack, I think we might lose some up-to-date information. We sent an invite request to contributors' email a month ago already.

**Comment by NikkiWines** [comment]
Ah, if you haven't been invited yet can you resend the email? It may have somehow gotten lost, someone on our contributor-management team will add you to slack. 

Going to push up a revert of this and then revert the revert so that the same changes are implemented but the commits are signed.

**Pull request merged**